### PR TITLE
[custom channels]: Aux signer batching fixes

### PR DESCRIPTION
**Pull request by guggero**
Depends on https://github.com/lightningnetwork/lnd/pull/9196.

This is a rebased (and slightly changed) version of https://github.com/lightningnetwork/lnd/pull/9074 that was already reviewed and merged into the `0-19-staging` branch.

The main difference to the staging branch version is that we use a `fn.ContextGuard` instead of a bare context in the link.

cc @jharveyb

### CODE REVIEW

File: htlcswitch/link.go [hunk 1]
```diff
@@ -387,8 +387,7 @@ type channelLink struct {
 	// our next CommitSig.
 	incomingCommitHooks hookMap
 
-	wg   sync.WaitGroup
-	quit chan struct{}
+	*fn.ContextGuard
```
Comment (jharveyb) [inline htlcswitch/link.go:390]: nit: add godoc comment

**Review by jharveyb — APPROVED** [review]
LGTM 👍🏽 

AFAICT the propagation of quit signals matches the original PR (and the previous behavior), and `ContextGuard` def makes sense here.

File: htlcswitch/link.go [hunk 2]
```diff
@@ -903,8 +905,10 @@ func (l *channelLink) syncChanStates() error {
 		// We've just received a ChanSync message from the remote
 		// party, so we'll process the message  in order to determine
 		// if we need to re-transmit any messages to the remote party.
+		ctx, cancel := l.WithCtxQuitNoTimeout()
```
Comment (Roasbeef) [inline htlcswitch/link.go:908]: 👍

**Review by Roasbeef — APPROVED** [review]
LGTM 🌙

**Pull request merged**